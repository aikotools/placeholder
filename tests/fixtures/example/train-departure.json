{
  "testId": "{{gen:uuid:test-train-departure-001}}",
  "testcase": "TC-TRAIN-DEPARTURE",
  "train": {
    "number": "{{gen:zugnummer:4837}}",
    "type": "RGE",
    "operator": "DB",
    "route": {
      "from": "Berlin Hbf",
      "to": "Hamburg Hbf"
    }
  },
  "departure": {
    "station": "Berlin Hbf",
    "platform": "{{gen:number:7}}",
    "scheduledTime": "{{time:calc:0:seconds}}",
    "actualTime": "{{time:calc:120:seconds}}",
    "delaySeconds": "{{gen:number:120}}",
    "date": "{{time:calc:0:dd.MM.yyyy}}",
    "time": "{{time:calc:0:HH\\:mm\\:ss}}"
  },
  "arrival": {
    "station": "Hamburg Hbf",
    "estimatedTime": "{{time:calc:5400:seconds}}",
    "estimatedDate": "{{time:calc:5400:dd.MM.yyyy}}",
    "estimatedTimeFormatted": "{{time:calc:5400:HH\\:mm\\:ss}}"
  },
  "passengers": {
    "count": "{{gen:number:142}}",
    "vip": "{{gen:boolean:true}}",
    "manifest": [
      {
        "id": "{{gen:uuid:passenger-001}}",
        "seat": "{{gen:number:15}}"
      },
      {
        "id": "{{gen:uuid:passenger-002}}",
        "seat": "{{gen:number:16}}"
      },
      {
        "id": "{{gen:uuid:passenger-003}}",
        "seat": "{{gen:number:17}}"
      }
    ]
  },
  "metadata": {
    "createdAt": "{{time:calc:0:yyyy-MM-dd\\:HH\\:mm\\:ss}}",
    "createdTimestamp": "{{time:calc:0:seconds}}",
    "version": "{{gen:number:1}}",
    "environment": "test",
    "testRunId": "{{gen:uuid:run-12345}}"
  },
  "checkpoints": [
    {
      "name": "Departure Check",
      "timestamp": "{{time:calc:0:seconds}}",
      "status": "{{gen:string:passed}}"
    },
    {
      "name": "Mid-Journey Check",
      "timestamp": "{{time:calc:2700:seconds}}",
      "status": "{{gen:string:passed}}"
    },
    {
      "name": "Arrival Check",
      "timestamp": "{{time:calc:5400:seconds}}",
      "status": "{{gen:string:passed}}"
    }
  ],
  "filename": "train_{{gen:zugnummer:4837}}_{{time:calc:0:yyyy-MM-dd}}_{{gen:uuid:run-12345}}.json"
}
