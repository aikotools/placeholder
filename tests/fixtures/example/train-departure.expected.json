{
  "testId": "test-train-departure-001",
  "testcase": "TC-TRAIN-DEPARTURE",
  "train": {
    "number": 4837,
    "type": "RGE",
    "operator": "DB",
    "route": {
      "from": "Berlin Hbf",
      "to": "Hamburg Hbf"
    }
  },
  "departure": {
    "station": "Berlin Hbf",
    "platform": 7,
    "scheduledTime": 1742040000,
    "actualTime": 1742040120,
    "delaySeconds": 120,
    "date": "15.03.2025",
    "time": "12:00:00"
  },
  "arrival": {
    "station": "Hamburg Hbf",
    "estimatedTime": 1742045400,
    "estimatedDate": "15.03.2025",
    "estimatedTimeFormatted": "13:30:00"
  },
  "passengers": {
    "count": 142,
    "vip": true,
    "manifest": [
      {
        "id": "passenger-001",
        "seat": 15
      },
      {
        "id": "passenger-002",
        "seat": 16
      },
      {
        "id": "passenger-003",
        "seat": 17
      }
    ]
  },
  "metadata": {
    "createdAt": "2025-03-15:12:00:00",
    "createdTimestamp": 1742040000,
    "version": 1,
    "environment": "test",
    "testRunId": "run-12345"
  },
  "checkpoints": [
    {
      "name": "Departure Check",
      "timestamp": 1742040000,
      "status": "passed"
    },
    {
      "name": "Mid-Journey Check",
      "timestamp": 1742042700,
      "status": "passed"
    },
    {
      "name": "Arrival Check",
      "timestamp": 1742045400,
      "status": "passed"
    }
  ],
  "filename": "train_4837_2025-03-15_run-12345.json"
}
